src/main/java/upbrella/be/store/service/ClassificationService.java

```diff
@@ -0,0 +1,84 @@
+package upbrella.be.store.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import upbrella.be.store.dto.request.CreateClassificationRequest;
+import upbrella.be.store.dto.request.CreateSubClassificationRequest;
+import upbrella.be.store.dto.response.AllClassificationResponse;
+import upbrella.be.store.dto.response.AllSubClassificationResponse;
+import upbrella.be.store.dto.response.SingleClassificationResponse;
+import upbrella.be.store.dto.response.SingleSubClassificationResponse;
+import upbrella.be.store.entity.Classification;
+import upbrella.be.store.repository.ClassificationRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class ClassificationService {
+
+    private final ClassificationRepository classificationRepository;
+
+    public Classification createClassification(CreateClassificationRequest request) {
+        return classificationRepository.save(ofCreateClassification(request));
+    }
+
+    public Classification createSubClassification(CreateSubClassificationRequest request) {
+        return classificationRepository.save(ofCreateSubClassification(request));
+    }
+
+    public void deleteClassification(Long id) {
+        classificationRepository.deleteById(id);
+    }
+
+    public AllClassificationResponse findAllClassification(String type) {
+        List<Classification> allByClassification = classificationRepository.findAllByClassification(type);
+        List<SingleClassificationResponse> classifications = new ArrayList<>();
+
+        for (Classification classification : allByClassification) {
+            classifications.add(SingleClassificationResponse.builder()
+                    .id(classification.getId())
+                    .name(classification.getName())
+                    .latitude(classification.getLatitude())
+                    .longitude(classification.getLongitude())
+                    .build());
+        }
+
+        return AllClassificationResponse.builder()
+                .classifications(classifications)
+                .build();
```

Comment:
```suggestion
        return AllClassificationResponse.fromClassifications(classificationRepository.findAllByClassification(type));
```
위처럼 mapping 로직을 DTO의 정적 팩터리 메서드로 빼는건 어떨까요?

Reply:
수정하겠습니다